lib/forEach.js

```diff
@@ -17,9 +17,10 @@ class ObjectToReadable extends Readable {
 }
 
 class ForEach extends Transform {
-  constructor (pipeline, master, handleChunk) {
+  constructor (pipeline, master, log, handleChunk) {
```

Comment:
Do we always have a Logger or should it be possible to use the logging feature optional (in the future)? Then I would prefer an interface like this: `constructor (pipeline, master, handleChunk, { log } = {})`.

Reply:
I think there would always be a logger instance, albeit hypothetical disabling would replace it with a `/dev/null`-esque. Otherwise we'd end up with `if (log)` scattered in multiple places.

Reply:
I agree that we would use some kind of dummy logger and don't put `if(log)` everywhere. But can we anyway change the params? If we would make it optional later, it would already fit into the logic of optional params als destructure param.